components/ws-manager/pkg/manager/internal/grpcpool/pool.go

```diff
@@ -98,3 +102,25 @@ func (p *Pool) Close() error {
 
 	return nil
 }
+
+// ValidateConnections check if existing connections in the pool
+// are using valid addresses and remove them from the pool if not.
+func (p *Pool) ValidateConnections() {
```

Comment:
Why not do this on every Get() call directly? This way we're not "wasting" five minutes until we identify invalid addresses.

Reply:
I tried that but introduces some delay (up to two seconds). If that's ok I can do that